component/attr_cache/attr_cache.go

```diff
@@ -210,42 +190,29 @@ func (ac *AttrCache) deletePath(path string, time time.Time) {
 func (ac *AttrCache) deleteCachedDirectory(path string, time time.Time) error {
 
 	// Delete all descendants of the path, then delete the path
-	// For example, filesystem: a/, a/b, a/c, aa/, ab.
-	// When we delete directory a, we only want to delete a/, a/b, and a/c.
-	// If we do not conditionally extend a, we would accidentally delete aa/ and ab
 
-	// Add a trailing '/' so that we only delete child paths under the directory and not paths that have the same prefix
-	prefix := dirToPrefix(path)
-	// remember whether we actually found any contents
-	foundCachedContents := false
-	for key, value := range ac.cacheMap {
-		if strings.HasPrefix(key, prefix) {
-			foundCachedContents = true
-			value.markDeleted(time)
-		}
-	}
+	//get attrCacheItem
+	toBeDeleted, err := ac.cacheMap.get(path)
 
-	// check if the directory to be deleted exists
-	if !foundCachedContents && !ac.pathExistsInCache(path) {
-		log.Err("AttrCache::deleteCachedDirectory : directory %s does not exist in attr cache.", path)
+	// delete the path itself and children.
+	if err != nil {
+		log.Err("could not find the cache map item due to the following error: ", err)
 		return syscall.ENOENT
+	} else {
+		toBeDeleted.markDeleted(time)
 	}
 
-	// We need to delete the path itself since we only handle children above.
-	ac.deletePath(path, time)
-
-	// If this leaves the parent or any ancestor directory empty, record that.
-	// Although this involves an unnecessary second traversal through the cache,
-	// because of the code complexity, I think it's worth the readability gained.
 	ac.updateAncestorsInCloud(getParentDir(path), time)
-
 	return nil
 }
 
 // pathExistsInCache: check if path is in cache, is valid, and not marked deleted
 func (ac *AttrCache) pathExistsInCache(path string) bool {
-	value, found := ac.cacheMap[internal.TruncateDirName(path)]
-	return (found && value.valid() && value.exists())
+	value, err := ac.cacheMap.get(internal.TruncateDirName(path))
+	if err != nil {
+		return false
+	}
+	return (value.valid() && value.exists())
```

Comment:
```suggestion
	return value.exists()
```
Now that `exists()` checks valid too, we can remove the `valid()` check from next to all the `exists()` calls everywhere.